lib/config/lib/params.c

```diff
@@ -542,17 +542,17 @@ param_default_jsonify(const struct param_spec *const ps, const void *const value
         case PARAM_TYPE_I32:
             return cJSON_CreateNumber(*(int32_t *)value);
         case PARAM_TYPE_I64:
-            return cJSON_CreateNumber(*(int64_t *)value);
+            return cJSON_CreateNumber((double)(*(int64_t *)value));
         case PARAM_TYPE_U8:
             return cJSON_CreateNumber(*(uint8_t *)value);
         case PARAM_TYPE_U16:
             return cJSON_CreateNumber(*(uint16_t *)value);
         case PARAM_TYPE_U32:
             return cJSON_CreateNumber(*(uint32_t *)value);
         case PARAM_TYPE_U64:
-            return cJSON_CreateNumber(*(uint64_t *)value);
+            return cJSON_CreateNumber((double)(*(uint64_t *)value));
         case PARAM_TYPE_SIZE:
-            return cJSON_CreateNumber(*(size_t *)value);
+            return cJSON_CreateNumber((double)(*(size_t *)value));
```

Comment:
What's the warning here?  Loss of precision?  It's kinda scary that we just accept it and pretend it didn't happen.  Will be nice to get away from cJSON...

Reply:
This is just cJSON garbage that will be gone when we switch to another JSON library. The warning is loss of precision or something of that nature.